string log_msg
---
string return